File: web/src/types/flipbook.ts
/**
 * 3D 翻页画册相关类型定义
 */

/** 画册页面数据 */
export interface FlipbookPage {
  id: string
  image?: string
  alt: string
  sourceFilename?: string
  width?: number
  height?: number
  caption?: string
  text?: string
  fit?: 'fill' | 'cover' | 'contain'
  padding?: number
  pageId?: string
  pageOrder?: number
  imageUrl?: string
  sourceImageId?: string
  imageWidth?: number | null
  imageHeight?: number | null
}

/** AI 生成的主题配置 */
export interface FlipbookTheme {
  mood?: string
  pageColor?: string
  coverColor?: string
  backCoverColor?: string
  pageTexture?: string
}

/** 画册项目 */
export interface FlipbookProject {
  projectId: string
  title: string
  kicker?: string
  status: 'creating' | 'analyzing' | 'ready' | 'error'
  coverUrl?: string | null
  themeJson?: string | FlipbookTheme | null
  pageCount: number
  pages: FlipbookPage[]
  createdAt: string
  updatedAt: string
  errorMessage?: string | null
}

/** 创建画册请求 */
export interface CreateFlipbookRequest {
  title: string
  resultIds: string[]
  kicker?: string
}

/** 画册布局参数 */
export interface BookLayout {
  width: number
  height: number
  minWidth: number
  minHeight: number
  maxWidth: number
  maxHeight: number
  defaultFit: 'fill' | 'contain'
  defaultPadding: number
}

File: server/app/schemas/flipbook.py
"""
3D 翻页画册 Schema
"""
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class FlipbookPageRead(BaseModel):
    """画册页面读取响应"""
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    page_id: str
    page_order: int
    image_url: str | None = None
    source_image_id: str | None = None
    image_width: int | None = None
    image_height: int | None = None
    alt: str | None = None
    caption: str | None = None
    text: str | None = None
    fit: str | None = None


class FlipbookProjectRead(BaseModel):
    """画册项目读取响应"""
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    project_id: str
    title: str
    kicker: str | None = None
    status: str
    cover_url: str | None = None
    theme_json: str | None = None  # AI生成的主题配置
    page_count: int
    error_message: str | None = None  # 错误信息
    pages: list[FlipbookPageRead] = []
    created_at: datetime
    updated_at: datetime


class FlipbookProjectBrief(BaseModel):
    """画册项目简要信息（用于列表）"""
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    project_id: str
    title: str
    kicker: str | None = None
    status: str
    cover_url: str | None = None
    page_count: int
    created_at: datetime


class FlipbookListResponse(BaseModel):
    """画册列表响应"""
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    total: int
    page: int
    page_size: int
    items: list[FlipbookProjectBrief]


class CreateFlipbookRequest(BaseModel):
    """创建画册请求"""
    model_config = ConfigDict(alias_generator=to_camel, populate_by_name=True)

    title: str = "Photo Book"
    kicker: str | None = "Folio"
    result_ids: list[str]  # 转换结果ID列表（从 style_results 中选取）
